src/supervisor.rs

```diff
@@ -87,152 +92,97 @@ async fn terminate_prev_siblings<'a, 'b, 'c>(
-                            .await;
-                        // append the worker to the started workers vector
-                        started_workers.push(started_worker);
-                        // keep pending_count up to date
-                        let pending_count: usize = pending_count - 1;
-                        //
-                        StartResult::SuccessfulStart {
-                            started_workers,
-                            pending_count,
-                        }
-                    }
-                    // when the child_spec failed to start
-                    Err((failed_worker_spec, start_error0)) => {
-                        let child_runtime_name =
-                            format!("{}/{}", sup_runtime_name, failed_worker_spec.name);
-                        let start_error = Arc::new(start_error0);
-                        // send event notification indicating that the worker failed to start
-                        ev_notifier
-                            .worker_start_failed(child_runtime_name, start_error.clone())
-                            .await;
-                        // transform current start to FailedStart
-                        StartResult::FailedStart {
-                            failed_worker_spec,
-                            start_error,
-                            // store the already started workers so that we can
-                            // later terminate them
-                            prev_started_workers: started_workers,
-                            // use the updated pending_count entry to get the
-                            // remaining pending_workers vec size
-                            pending_workers: Vec::with_capacity(pending_count),
-                        }
-                    }
-                }
-            }
-            // If we already have a failed start, we just append entries to the
-            // pending_workers vec
-            StartResult::FailedStart {
-                failed_worker_spec,
-                start_error,
-                prev_started_workers,
-                mut pending_workers,
-            } => {
-                pending_workers.push(child_spec);
-                StartResult::FailedStart {
-                    failed_worker_spec,
-                    start_error,
-                    prev_started_workers,
-                    pending_workers,
-                }
-            }
+        // when the child_spec failed to start
+        Err((failed_worker_spec, start_error0)) => {
+            let child_runtime_name = format!("{}/{}", sup_runtime_name, failed_worker_spec.name);
+            let start_error = Arc::new(start_error0);
+            // send event notification indicating that the worker failed to start
+            ev_notifier
+                .worker_start_failed(child_runtime_name, start_error.clone())
+                .await;
+            Err((failed_worker_spec, start_error))
```

Comment:
Wow, this is dramatically smaller! So much easier to understand too. I was hoping it would be an improvement but this really surprised me.